src/hevm/src/EVM/Expr.hs

```diff
@@ -672,6 +672,26 @@ simplify e = if (mapExpr go e == e)
       | x == 0 = b
       | otherwise = a
 
+    -- div/sdiv/mod/smod by zero
+    go (Div _ (Lit 0)) = Lit 0
+    go (Mod _ (Lit 0)) = Lit 0
+    go (SDiv _ (Lit 0)) = Lit 0
+    go (SMod _ (Lit 0)) = Lit 0
```

Comment:
That's a great idea! Let me do that.

I think I'll leave them here too just in case we have something simplified to 0 during `simplify` -- not sure that's a perfect idea, let me know what you think.

Reply:
So it turns out the div/mod/etc. in `Expr` all handle the 0 case already, but only if both are concrete. I don't think it's very often we'll see one being 0, the other being non-concrete. So I removed this, it's not that important. In my case, it was important, because it guarded against division by zero etc. That's no longer needed if we use div/mod/etc from `Expr`.